devicetypes/smartthings/zwave-siren-tamper.src/zwave-siren-tamper.groovy

```diff
@@ -0,0 +1,283 @@
+
+	response([secure(zwave.basicV1.basicGet()), zwave.batteryV1.batteryGet()])
+}
+
+def updated() {
+	def commands = []
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	if (isYale()) {
+
+		if (!state.alarmLength) state.alarmLength = 10 // default value
+		if (!state.alarmLEDflash) state.alarmLEDflash = 1 // default value
+		if (!state.comfortLED) state.comfortLED = 0 // default value
+		if (!state.tamper) state.tamper = 0 // default value
+
+		log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+		Short alarmLength = (settings.alarmLength as Short) ?: 1
+		Boolean alarmLEDflash = (settings.alarmLEDflash as Boolean) ?: 0
+		Short comfortLED = (settings.comfortLED as Short) ?: 0
+		Boolean tamper = (settings.tamper as Boolean) ?: 0
+
+		if (alarmLength != state.alarmLength || alarmLEDflash != state.alarmLEDflash || comfortLED != state.comfortLED || tamper != state.tamper) {
+			state.alarmLength = alarmLength
+			state.alarmLEDflash = alarmLEDflash
+			state.comfortLED = comfortLED
+			state.tamper = tamper
+
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 1, size: 1, configurationValue: [alarmLength]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 2, size: 1, configurationValue: [alarmLEDflash ? 1 : 0]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 3, size: 1, configurationValue: [comfortLED]).format())
+			commands << new physicalgraph.device.HubAction(zwave.configurationV1.configurationSet(parameterNumber: 4, size: 1, configurationValue: [tamper ? 1 : 0]).format())
+			commands << "delay 1000"
+			commands << secure(zwave.basicV1.basicSet(value: 0x00))
+		}
+	}
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (state.sec) {
+			result = createEvent(descriptionText: description, displayed: false)
+		} else {
+			result = createEvent(
+				descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+				eventType: "ALERT",
+				name: "secureInclusion",
+				value: "failed",
+				displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x59: 1, 0x2B: 1, 0x71: 2, 0x85: 2, 0x25: 2, 0x7A: 1])
```

Comment:
Can we use notification v3 (`0x71: 3`) instead of v2?

Reply:
They are backwards compatible with each other, but you may need to change how some things are named.